turbinia/state_manager.py

```diff
@@ -267,77 +247,127 @@ def get_task_data(
+    # If no task_id or request_id is passed, gets all valid saved tasks
+    else:
+      task_ids = [task_key.split(':')[1] for task_key in self.iterate_keys('Task')]
+
+    tasks = []
 
+    for task_id in task_ids:
+      task = self.get_task(task_id)
+      if self.validate_task(task, instance, days, group_id, user):
+        tasks.append(task)
+    
     return tasks
 
+  def format_task(self, task):
+    task_dict = self.get_task_dict(task)
+    task_dict['last_update'] = task_dict['last_update'].strftime(
+        DATETIME_FORMAT)
+    task_dict['start_time'] = task_dict['start_time'].strftime(DATETIME_FORMAT)
+    if not task_dict.get('status'):
+      task_dict['status'] = (
+        f'Task scheduled at {datetime.now().strftime(DATETIME_FORMAT)}')
+    if task_dict['run_time']:
+      task_dict['run_time'] = task_dict['run_time'].total_seconds()
+    for key, value in task_dict.items():
+      try:
+        task_dict[key] = json.dumps(value)
+      except (TypeError, ValueError) as exception:
+        error_message = f'Error serializing task attribute for task {task.id}.'
+        log.error(f'{error_message}: {exception}')
+        raise TurbiniaException(error_message) from exception
+    return task_dict
+
+  def update_request_task(self, task):
+    request_key = ':'.join(('TurbiniaRequest', task.request_id))
+    self.add_to_list(request_key, 'task_ids', task.id)
+    request_last_update = datetime.strptime(self.get_attribute(
+      request_key, 'last_update'), DATETIME_FORMAT)
+    try:
+      last_update = json.dumps(max(request_last_update, task.last_update).strftime(
+        DATETIME_FORMAT))
+    except redis.RedisError as exception:
+      error_message = f'Error encoding key {request_key} in Redis'
+      log.error(f'{error_message}: {exception}')
+      raise TurbiniaException(error_message) from exception   
+    self.set_attribute(request_key, 'last_update',last_update)
+    statuses_to_remove = ['succesful_tasks', 'failed_tasks','running_tasks', 'queued_tasks']
+    # 'successful' could be None or False, which means different things.
+    # If False, the task has failed, If None, could be queued or running.
+    if hasattr(task, 'succesful'):
+      if task.successful:
+        self.add_to_list(request_key, 'succesful_tasks', task.id)
+        statuses_to_remove.remove('succesful_tasks')
+      if task.successful is False:
+        self.add_to_list(request_key, 'failed_tasks', task.id)
+        statuses_to_remove.remove('failed_tasks')
+      elif task.successful is None:
+        if task.status:
+          if 'running' in task.status:
+            self.add_to_list(request_key, 'running_tasks', task.id)
+            statuses_to_remove.remove('running_tasks')
+        else:
+          # 'successful' is None and 'status' is None
+          self.add_to_list(request_key, 'queued_tasks', task.id)
+          statuses_to_remove.remove('queued_tasks')
+    for status_name in statuses_to_remove:
```

Comment:
Hey @Igor8mr do you remember why we run this for loop here? It looks like we're adding to the list above, depending on the value of task.successful but then the for loop removes all of them?

Reply:
Hey @jleaniz, line 308 initializes a list with all status's names (statuses_to_remove), and the if statements on lines 312, 315, 318, and 323 will check the right status and remove its name from the list. So, after that, the list will have all statuses, except the right one, so we can remove the task from its old status list on Redis after we add it to the right one. This statuses_to_remove list is only so we can keep track of which lists in Redis must be updated to remove the task from there, as we don't want to remove it from the list we just added it to. The for loop iterates over all wrong statuses to check which one has the task and remove it. I think I didn't find a way to just go to the right one and remove it, but there might be a better solution for that. Let me know if that isn't clear or if you have more questions about the code. Thanks for working on it!